src/components/links.jsx

```diff
@@ -40,50 +53,44 @@ const LinksView = Backbone.View.extend({
 
     // text placeholders for all edges
     newLinksG
-      .append('text')
-      .attr('x', d => d.source.x)
-      .attr('y', d => d.source.y)
+      .append('text');
+    container.selectAll('.link').selectAll('text')
+      .attr('x', d => d.source.x || 0)
+      .attr('y', d => d.source.y || 0)
       .attr('text-anchor', 'middle')
       .text(() => ' ');
 
     // double and triple bonds
     newLinksG
       .filter(d => d.bond > 1)
       .append('line')
-      .attr('class', 'separator')
+      .attr('class', 'separator separator-double');
+    container.selectAll('.separator-double')
       .style('stroke', '#FFF')
       .style('stroke-width', d => `${(d.bond * 4) - 5}px`);
 
     // triple bonds
     newLinksG
       .filter(d => d.bond === 3)
       .append('line')
-      .attr('class', 'separator')
+      .attr('class', 'separator separator-triple');
+    container.selectAll('.separator-triple')
       .style('stroke', d => molViewUtils.chooseColor(d, 'black'))
       .style('stroke-width', () => molViewUtils.getBondWidth(1));
 
-    return this;
-  },
+    links.exit()
+      .remove();
+  }
 
-  renderPosition() {
-    const links = this.svg.selectAll('.link');
-
-    // keep edges pinned to their nodes
-    links.selectAll('line')
-      .attr('x1', d => d.source.x)
-      .attr('y1', d => d.source.y)
-      .attr('x2', d => d.target.x)
-      .attr('y2', d => d.target.y);
+  render() {
+    return (
+      <g className="links-container" ref={(c) => { this.linksContainer = c; }} />
```

Comment:
you should check if c is defined. sometimes its not (or at least in v14 i had this problem)